### PR TITLE
Setup creme with pip install

### PR DESCRIPTION
**Pull request by hsmett**
This pull request changes the way to install creme and it's requirements.

`pip install .[dev,mysql,pgsql,graphs]` will setup creme directly in the current python environment.
`pip install -e .[dev,mysql,pgsql,graphs]` (develop mode) can be used to keep the former behaviour.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -4,30 +4,50 @@ MAKE_NPROCS ?= $(shell nproc)
...
 
 
 ## Run the Django test suite
 .PHONY: test
 test:
-	python manage.py test --noinput --parallel=${MAKE_NPROCS} $(filter-out $@,$(MAKECMDGOALS))
+	python creme/manage.py test --noinput --parallel=${MAKE_NPROCS} $(filter-out $@,$(MAKECMDGOALS))
 
 
 ## Run the Django test suite and generate coverage reports
```
Comment (joehybird) [inline Makefile:53]: The test-cov target needs to be updated too :)

File: .circleci/config.yml [hunk 2]
```diff
@@ -167,10 +165,10 @@ jobs:
       - install-py-dev-env
       - run: cp ~/project/.circleci/circleci_settings.py creme/project_settings.py
       - run: cp ~/project/.circleci/sqlite3_settings.py creme/local_settings.py
-      - run: python manage.py migrate
-      - run: python manage.py creme_populate
-      - run: python manage.py generatemedia
-      - run: COVERAGE_PROCESS_START=setup.cfg coverage run --source creme/ manage.py test --noinput --parallel=8
+      - run: python creme/manage.py migrate
```
Comment (joehybird) [inline .circleci/config.yml:168]: `- run: MAKE_NPROCS=8 make test-cov` ?

File: setup.cfg [hunk 3]
```diff
@@ -25,18 +25,36 @@ classifiers =
...
+    xlrd >=1.2.0,<1.2.1024
+    xlwt >=1.3.0,<1.3.1024
+
+[options.extras_require]
+dev=
+    coverage >=5.1,<5.1.1024
+    flake8 >=3.8.2,<3.8.1024
+    isort >=4.3.21,<4.3.1024
+    parameterized >=0.7.1,<0.7.1024
+    tblib >=1.6.0,<1.6.1024
```
Comment (joehybird) [inline setup.cfg:49]: please add `django-extensions` :pray:

File: .circleci/config.yml [hunk 4]
```diff
@@ -26,17 +26,15 @@ commands:
     steps:
       - restore_cache:
           keys:
-            - creme_crm-{{ checksum "creme/requirements.txt" }}-{{ checksum "creme/requirements-dev.txt" }}-{{ checksum "creme/requirements-mysql.txt" }}-{{ checksum "creme/requirements-pgsql.txt" }}
-            - creme_crm-{{ checksum "creme/requirements.txt" }}-{{ checksum "creme/requirements-dev.txt" }}
+            - creme_crm-{{ checksum "setup.cfg" }}
```
Comment (joehybird) [inline .circleci/config.yml:29]: :+1:

File: .circleci/config.yml (hunk 2 above)
Comment (joehybird) [reply .circleci/config.yml:168]: By the way I am not sure that the `coverage html` is really useful here... `coverage report` seems more useful.

Comment (hsmett) [reply .circleci/config.yml:168]: coverage html generates html reports that are then saved as build artifacts, and can be browse.
Ex: https://1442-264866391-gh.circle-artifacts.com/0/artifacts/coverage_html/index.html

File: setup.cfg (hunk 3 above)
Comment (hsmett) [reply setup.cfg:49]: I agree it's really usefull and we should add it, but it's not the purpose of this PR

File: .circleci/config.yml (hunk 2 above)
Comment (joehybird) [reply .circleci/config.yml:168]: Oh ok, I though it is was lost in the space. Can you print the link ?

Comment (hsmett) [reply .circleci/config.yml:168]: It already printed, look :D ! 
image

File: setup.cfg (hunk 3 above)
Comment (genglert) [reply setup.cfg:49]: This line is already in the current requirements-dev.txt:
django-extensions >=2.2.9,<2.2.1024

**Comment by genglert** [comment]
README still references "requirements*.txt" ; please remove these references & explain instead how to install with 'pip'.

File: setup.cfg [hunk 5]
```diff
@@ -25,18 +25,36 @@ classifiers =
...
 packages = find:
+install_requires =
+    django >=2.2.10,<2.2.1024
+    redis >=3.4.1,<3.4.1024
+    Pillow >=7.0.0,<7.0.1024
+    python-dateutil >=2.8.1,<2.8.1024
+    bleach >=3.1.1,<3.1.1024
+    django-formtools ==2.2
+    xlrd >=1.2.0,<1.2.1024
+    xlwt >=1.3.0,<1.3.1024
```
Comment (genglert) [inline setup.cfg:41]: Beware when rebasing some upgrades have been done ( django >=2.2.15, Pillow >=7.2.0, bleach >=3.1.5, )

**Review by joehybird — APPROVED** [review]
:rocket:

File: doc/fr/Didacticiel - Coder un module.rst [hunk 6]
```diff
@@ -242,13 +242,13 @@ Créer la table dans la base de données
 
 Toujours depuis le répertoire ``creme/``, lancez les commandes suivantes : ::
```
Comment (genglert) [inline doc/fr/Didacticiel - Coder un module.rst:243]: Update the date in the header.

**Review by genglert — APPROVED** [review]

**Pull request merged**